(doc)comment nits

Niki4tap opened the pull request:
@xFrednet

![image](https://user-images.githubusercontent.com/66798058/210443643-bf71b49a-3d71-4f88-89a1-d5131136cfac.png)

Whew, that turned out way bigger than I thought it would be, feel free to take your time on this one, or just ignore until exams pass :sweat_smile: 

Even though this changes a lot of files, the changes themselves are very small (and mostly affect only (doc)comments).

I tried to touch only the (doc)comments, and not the actual code, but I think there is some weirdness with function names in `marker_api/src/ast/item/adt_item.rs`, and that *might be a little bit of a breaking change to the public API*, but I don't think we care yet?

Some of these are just my personal nits, so feel free to disagree or just not accept them.

xFrednet commented:
You really read through most of the code base, I'm really impressed. If you have any suggestions regarding the project structure or anything else, you're very welcome to share them. This is the first Rust project I'm structuring myself, and also the first AST design ^^

I'll wait with reviewing this until after the exams, so it should be this weekend. This is a gigantic help <3

xFrednet commented:
> that might be a little bit of a breaking change to the public API, but I don't think we care yet?

No, we aren't concerned with that yet. AFAIK, the project is not used by anyone yet, so we can break everything as much as we want to. I regularly do that ^^. Even with the first dev versions, so v0.0.1 once all nodes and a basic backend is added, we can still break the API, though we might want to keep a changelog at that time.

Defining a stability guarantee, will be on the list for v1.0.0. The goal is generally to have the most stable API possible, but I think we should define if and under which circumstances breaking changes would be allowed. Maybe we can also have an edition model, like rustc, using features? The awesome part, is that this is all up to us to decide, when the time comes :)

Niki4tap commented:
note: this PR got split into 3:
- This PR still has (doc)comment nits and other fixups
- #78 with API changes in ADT items
- #77 with leftovers from #71

sorry for the mess :)

Niki4tap commented:
I'll resolve the conflict in a bit, but I think it's good for review as-is.

xFrednet commented on marker_api/src/ast/common/callable.rs:
```diff
@@ -22,15 +22,15 @@ pub trait CallableData<'ast> {
     /// only use the `extern` keyword to specify the ABI. These will currently
     /// still return `false` even if the keyword is present. In those cases,
-    /// please refer to the ABI instead.
+    /// please refer to the [`abi()`][Self::abi] instead.
```
Do you want to adjust this to use brackets for the link target? Seeing this, I wonder if this is worth a clippy lint, that unifies all link types ^^

Niki4tap replied:
Sure, I wonder if `rustfmt` can format these

xFrednet replied:
IDK, rustfmt, usually avoids doc comments, as markdown formatting can be complicated. Clippy has a few lints that parse the markdown. With that, it should be possible :)

xFrednet replied:
https://github.com/rust-lang/rust-clippy/issues/10176

xFrednet commented on marker_api/src/ast/ty.rs:
```diff
@@ -97,26 +97,58 @@ pub enum TyKind<'ast> {
-    /// A tuple type like `()`, `(T, U)`
+    /// A tuple type like [`()`][prim@tuple], [`(T, U)`][prim@tuple]
     Tuple(&'ast TupleTy<'ast>),
-    /// An array with a known size like: `[T; n]`
+    /// An array with a known size like: [`[T; N]`][prim@array]
     Array(&'ast ArrayTy<'ast>),
-    /// A variable length slice like `[T]`
+    /// A variable length slice like [`[T]`][prim@slice]
```
Also, these doc comments. The `prim@` link is actually interesting. Probably one of these things that very few projects run into. And I'm happy to be part of one of these projects :joy:

Niki4tap replied:
Haha, yeah, glad to see that my "random looking into stdlib source" pays off. A bit sad that there isn't a shortcut like this for keywords, only for primitive types :(

xFrednet reviewed (approved):
Damn, I made numerous typos :sweat_smile: In the browser etc, I use LanguageTool, which works pretty well (from what I can tell) and has some context awareness, sadly it doesn't integrate too well with IDEs :/. Thank you for fixing all of these, writing the documentation for v0.0.1 will be fun :sweat_smile:

I have two tiny nits related to links, as we just had the topic in #82. Otherwise, it's good to go. You can r=xFrednet after the links have been updated :)

bors d+

Niki4tap commented:
> writing the documentation for v0.0.1 will be fun

I'll try to catch any further typos at the PR stage :)

Niki4tap commented:
Updated the links to follow one style, also added a small nit I didn't notice earlier in #82, didn't want to make a separate PR for it, hope it's ok here as a separate commit :sweat_smile:

bors r=xFrednet

xFrednet commented:
Adding a new commit is totally fine. Thank you!
